src/master/master_heartbeat.cpp

```diff
@@ -1334,6 +1340,16 @@ hb_cluster_calc_score (void)
 	}
     }
 
+  cubhb::node_entry *new_master = hb_Cluster->master;
+  if (old_master != NULL && new_master != NULL && old_master->get_hostname () != new_master->get_hostname ())
```

Comment:
if the slave cub_server crashes, will it still receive the hostname after restart?

Reply:
valid point, fixed